File: creational-pattern/singleton-pattern/singleton.go
package singleton_pattern

type Singleton interface {
	AddOne() int
	GetCount() int
}

type singleton struct {
	count int
}

var instance *singleton

func GetInstance() Singleton {
	if instance == nil {
		return new(singleton)
	}
	return instance
}

func (s *singleton) AddOne() int {
	s.count += 1
	return s.count
}

func (s *singleton) GetCount() int {
	return s.count
}

File: creational-pattern/singleton-pattern/java_code/java_tut/singleton/multiThreaded/non_thread_safe/singleton.java
package singleton.multiThreaded.non_thread_safe;

public final class singleton {

    private static singleton instance;

    public static Object multiThreaded;

    public String value;

    private singleton(String value) {

        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {

            e.printStackTrace();
        }
        this.value = value;
    }

    public static singleton getInstance(String value) {
        if (instance == null) {
            System.out.println("creating new instance of singleton class");
            instance = new singleton(value);
        }
        return instance;
    }
}
